### ISSUE TITLE
[MSSQL] Incorrect syntax when sql statement is created by Liquibase

### ISSUE THREAD

**Issue by dalyodhran** [liquibase/liquibase#1158]
## Environment
Java 8

**Liquibase Version**:
3.6.3

**Liquibase Integration & Version**: 
Gradle

**Liquibase Extension(s) & Version**:
N/A

**Database Vendor & Version**:
MsSql 2017

**Operating System Type & Version**:
Ubuntu 16.04.3 LTS

## Description
When creating a table with column: 

```xml
<column name="REPRESENTATION" type="TEXT(25500)"/>
```

Liquibase converts `TEXT` to `varchar (max) (25500)`

## Steps To Reproduce
Use Liquibase with the following change log:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
    <changeSet author="bburke@redhat.com" id="1.3.0">
        <createTable tableName="ADMIN_EVENT_ENTITY">
            <column name="ID" type="VARCHAR(36)">
                <constraints nullable="false"/>
            </column>
            <column name="ADMIN_EVENT_TIME" type="BIGINT"/>
            <column name="REALM_ID" type="VARCHAR(255)"/>
            <column name="OPERATION_TYPE" type="VARCHAR(255)"/>
            <column name="AUTH_REALM_ID" type="VARCHAR(255)"/>
            <column name="AUTH_CLIENT_ID" type="VARCHAR(255)"/>
            <column name="AUTH_USER_ID" type="VARCHAR(255)"/>
            <column name="IP_ADDRESS" type="VARCHAR(255)"/>
            <column name="RESOURCE_PATH" type="VARCHAR(2550)"/>
            <column name="REPRESENTATION" type="TEXT(25500)"/>
            <column name="ERROR" type="VARCHAR(255)"/>
        </createTable>
    </changeSet>
</databaseChangeLog>
```

## Actual Behavior
```
Error: Incorrect syntax near '('. 
[Failed SQL: CREATE TABLE ADMIN_EVENT_ENTITY (
    ID varchar(36) NOT NULL,
    ADMIN_EVENT_TIME bigint,
    REALM_ID varchar(255), 
    OPERATION_TYPE varchar(255), 
    AUTH_REALM_ID varchar(255), 
    AUTH_CLIENT_ID varchar(255), 
    AUTH_USER_ID varchar(255), 
    IP_ADDRESS varchar(255), 
    RESOURCE_PATH varchar(2550), 
    REPRESENTATION varchar (max) (25500), 
    ERROR varchar(255)
)] 
```

## Expected/Desired Behavior
Table created in MsSql okay

┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-269) by [Unito](https://www.unito.io/learn-more)
┆Fix Versions: Liquibase 3.10.3

**Comment by molivasdat** [comment]
Hi @dalyodhran . Thanks for being very thorough with detailed description of the issue.  We will add this to our list of issues to process.

**Comment by molivasdat** [comment]
Hi @dalyodhran .  I created the PR in 4.0.x branch.  Is it ok to fix in 4.0.1 or do you need this in the 3.10.x branch?

**Comment by dalyodhran** [comment]
Hi @molivasdat. Thank you for creating a PR for this fix, we are currently using 3.10.x in our build, so it would be great if you could put the fix there. Thank you
